study1/week1/third-maximum-number.java

```diff
@@ -0,0 +1,37 @@
+class Solution {
+    public int thirdMax(int[] nums) {
+        //time complexity O(n)
+        //space complexiti O(1)
+
+        if (nums.length == 0 || nums == null) {
+            return 0; //or -1?
+        }
+        
+        long firstMax = Long.MIN_VALUE;
+        long secondMax = Long.MIN_VALUE;
+        long thirdMax = Long.MIN_VALUE;
+
+        for (int i : nums) { //nums = 321
+            // skip duplicated number 
+            if (i == firstMax || i == secondMax || i == thirdMax) {
```

Comment:
It's arguable, but these comparison will most likely be just an additional computation in case there's not much duplication in the array.

Could you try to think in a random generated array, how frequently would you think there would be a duplication in the array?

Reply:
so what I meant is that if you just delete this if statement, it will also pass. 
Also, you don't need to feel obligated to reply to my PR review comments. But of course if you have questions, you can still do.